pymc/sampling/forward.py

```diff
@@ -497,22 +499,100 @@ def sample_posterior_predictive(
+        getLogger("pymc").setLevel("INFO")
+
+        kwargs = {"progressbar": False, "random_seed": 0}
+
+        with pm.Model() as model:
+          x = pm.Normal("x")
+          y = pm.Normal("y")
+          z = pm.Normal("z", x + y**2)
+          det = pm.Deterministic("det", pm.math.exp(z))
+          obs = pm.Normal("obs", det, 1, observed=[20])
+
+          idata = pm.sample(tune=10, draws=10, chains=2, **kwargs)
+
+          # Default behavior
+          # Posterior predictive draws of `obs`, conditioned on the posterior draws of `z`
+          pm.sample_posterior_predictive(idata, var_names=["obs"], **kwargs)
+          # Sampling: [obs]
+
+          # Deterministic re-computation of `det` given the posterior draws of `z`
+          pm.sample_posterior_predictive(idata, var_names=["det"], **kwargs)
+          # Sampling: []
+
+          # Posterior predictive draws of `z` and `det`, conditioned on the posterior draws of `x` and `y`
+          # It does not matter that `z` was not an observed variable.
+          pm.sample_posterior_predictive(idata, var_names=["z", "det"], **kwargs)
+          # Sampling: [z]
 
-    Generate 5 posterior predictive samples per posterior sample.
+          # Posterior predictive draws of `y`, `z` and `det`, conditioned on the posterior draws of `x`
+          # Note that `y` samples are equivalent to the prior, since no other variables influence it.
+          pm.sample_posterior_predictive(idata, var_names=["y", "z", "det"], **kwargs)
+          # Sampling: [y, z]
+
+          # Same as before, except `z` is not stored in the returned trace.
+          # For computing `det` we still have to resample `z` as it depends on `y`, which is also being sampled.
+          pm.sample_posterior_predictive(idata, var_names=["y", "det"], **kwargs)
+          # Sampling: [y, z]
+
+          # Everything is sampled, which is equivalent to calling `sample_prior_predictive`
+          pm.sample_posterior_predictive(idata, var_names=["x", "y", "z", "det"], **kwargs)
+          # Sampling: [x, y, z]
+
+
+        # Notice that "sampling" a dependent deterministic doesn't force random variables to be sampled.
+        with pm.Model() as model:
+          x = pm.Normal("x")
+          y = pm.Normal("y")
+          det_xy = pm.Deterministic("det_xy", x + y**2)
+          z = pm.Normal("z", x + y**2)
+          det_z = pm.Deterministic("det_z", pm.math.exp(z))
+          obs = pm.Normal("obs", det, 1, observed=[20])
+
+          idata = pm.sample(tune=10, draws=10, chains=2, **kwargs)
+
+          pm.sample_posterior_predictive(idata, var_names=["det_xy", det_z"], **kwargs)
+          Sampling: [z]
+
+
+    You can manipulate the InferenceData to control the number of samples
 
     .. code:: python
+        import pymc as pm
+
+        with pm.Model() as model:
+            ...
+            idata = pm.sample()
+
+        # Generate 1 posterior predictive sample for every 5 posterior samples.
```

Comment:
TBH this example could just be removed. It has nothing to do with the behavior of this function, and we don't recommend thinning anyway.

Reply:
This was an pre-existing example in the documentation, it's a different use case and hence in a different block. Users were indeed asking how to reduce the number of draws in posterior_predictive when we took an argument away that allowed to do it. 

The thinning recommendation is for posterior sampling diagnostics, it's still fine to thin for doing predictions, specially if you have memory concerns